pkg/tfgen/docs.go

```diff
@@ -1129,6 +1146,111 @@ func (p *tfMarkdownParser) parseImports(subsection []string) {
 	}
 }
 
+// Recognizes import sections such as ones found in aws_accessanalyzer_analyzer. If the section is
+// recognized, patches up instructoins to make sense for the Pulumi projection.
+func tryParseV2Imports(typeToken string, markdownLines []string) (string, bool) {
+	var out bytes.Buffer
+	fmt.Fprintf(&out, "## Import\n\n")
+
+	markdown := strings.Join(markdownLines, "\n")
+	pn := parseNode(markdown)
+	if pn == nil {
+		return "", false
+	}
+
+	foundCode := false
+
+	for {
+		recognized := false
+		switch pn.Type {
+		case bf.CodeBlock:
+			code := string(pn.Literal)
+			switch string(pn.CodeBlockData.Info) {
+			case "terraform":
+				// Ignore terraform blocks such as:
+				//
+				// ```terraform
+				// import {
+				// 	to = aws_accessanalyzer_analyzer.example
+				// 	id = "example"
+				// }
+				// ```
+				recognized = true
+			case "console", "":
+				// Recognize import example codeblocks.
+				if ok, _ /* TFtype */, name, id := parseImportCode(code); ok {
+					emitImportCodeBlock(&out, typeToken, name, id)
+					recognized = ok
+					foundCode = true
+				}
+			}
+		case bf.Heading:
+			if pn.FirstChild != nil && pn.FirstChild.Type == bf.Text {
+				if string(pn.FirstChild.Literal) == "Import" {
+					// Skip "## Import" heading.
+					recognized = true
+				}
+			}
+		case bf.Paragraph:
+			// Propagate paragraphs to output.
+			paraMD, err := parseTextSeq(pn.FirstChild, false)
+			if err == nil {
+				fmt.Fprintf(&out, "%s\n\n", paraMD)
+				recognized = true
+			}
+		}
+		if !recognized {
+			return "", false
+		}
+		pn = pn.Next
+		if pn == nil {
+			break
+		}
+	}
+	if !foundCode {
+		return "", false
+	}
+	return out.String(), true
+}
+
+func emitImportCodeBlock(w io.Writer, typeToken, name, id string) {
+	fmt.Fprintf(w, "```sh<break>\n")
+	fmt.Fprintf(w, "$ pulumi import %s %s %s\n", typeToken, name, id)
+	fmt.Fprintf(w, "<break>```<break>\n")
+}
+
+var importCodePattern = regexp.MustCompile(
+	`^[%$] (?:pulumi|terraform) import` +
+		`[\\\s]+([^.]+)[.]([^\s]+)[\\\s]+([^\s]+)\s*$`)
+
+// Data correction: ignore the first of two acidentally repeated tokens such as:
+//
+//	% pulumi import aws_n_policy.example aws_n_policy.example id
+var importCodePattern2 = regexp.MustCompile(
+	`^[%$] (?:pulumi|terraform) import[\\\s]+[^\s]+` +
+		`[\\\s]+([^.]+)[.]([^\s]+)[\\\s]+([^\s]+)\s*$`)
+
+// Recognize import example codeblocks.
+//
+// I.E, given:
+//
+//	s := "% pulumi import aws_accessanalyzer_analyzer.example example"
+//
+// Returns this:
+//
+//	parseImportCode(s) == (true, "aws_accessanalyzer_analyzer", "example", "example")
+func parseImportCode(code string) (bool, string, string, string) {
```

Comment:
It's really difficult to understand what we are returning here. Can we add an annotation? Also, if the first `bool` is found, I think idiomatic go has it at the end.
```suggestion
func parseImportCode(code string) (token string, ??? string, ??? string, found bool) {
```

Reply:
Yup, fixed up.